src/tests/test_eigenio_matrixmarket.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2009-2020 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE eigenio_matrixmarket
+
+// Third party includes
+#include <boost/test/unit_test.hpp>
+
+// Local VOTCA includes
+#include "votca/tools/eigenio_matrixmarket.h"
+#include "votca/tools/votca_tools_config.h"
+
+using namespace votca::tools;
+
+BOOST_AUTO_TEST_SUITE(eigenio_matrixmarket)
+
+BOOST_AUTO_TEST_CASE(readvector_test) {
+
+  Eigen::VectorXd ref = Eigen::VectorXd::Zero(4);
+  ref << 1.0, 2.0, 3.0, 4.0;
+  std::cout << std::string(TOOLS_CMAKE_BINARY_DIR) << std::endl;
+  Eigen::VectorXd readin = EigenIO_MatrixMarket::ReadVector(
+      std::string(TOOLS_CMAKE_BINARY_DIR) +
```

Comment:
If you only read the files, why can you not read them from the `PROJECT_SOURCE_DIR`? No copy required.

Reply:
yes true, but what happens if I comile two different branches to two different build dirs?

Reply:
It will always use the current checkout of the files, which is the same behavior as for all the sources. I think the copy around of code/files makes stuff very confusing as you can have a source (`cc` files) and input file mismatch.

Reply:
maybe I do not understand correctly, how can there be a mismatch? The files are always copied when cmake is executed? 

My fear was the following scenario I have `build1` and `build2` which compiled two different branches, then symlink can point to files which are gone already.

You once mentioned symlink does not work on all systems? Is that still relevant?

Reply:
Well you build `branch1` and then cmake copies files over, then you checkout and build `branch2`, now you got back to the builddir of `branch1` and cmake will detect that the `cc` files have changes and recompiles stuff, e.g. the tests, but the copied files won't get updated until you rerun cmake.

You could of course copy the files a compile time, not cmake time (see the handling of `rst` files in `share/doc/sphinx`), but that seems a bit over the top.

And yes, symlinks are file system features and some file systems (mostly windows) don't support them.

Reply:
okay I am convinced.

Reply:
symlink it is.

Reply:
Or just read from `PROJECT_SOURCE_DIR`.